src/NuGet.Core/NuGet.ProjectModel/LockFile/Utf8JsonStreamLockFileTargetConverter.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Text.Json;
+using NuGet.Frameworks;
+
+namespace NuGet.ProjectModel
+{
+    /// <summary>
+    /// A <see cref="Utf8JsonStreamReaderConverter{T}"/> to allow read JSON into <see cref="LockFileTarget"/>
+    /// </summary>
+    internal class Utf8JsonStreamLockFileTargetConverter : Utf8JsonStreamReaderConverter<LockFileTarget>
+    {
+        public override LockFileTarget Read(ref Utf8JsonStreamReader reader)
+        {
+            if (reader.TokenType != JsonTokenType.PropertyName)
+            {
+                throw new JsonException("Expected PropertyName, found " + reader.TokenType);
+            }
+
+            var lockFileTarget = new LockFileTarget();
+            //We want to read the property name right away
+            var propertyName = reader.GetString();
+            var parts = propertyName.Split(JsonUtility.PathSplitChars, 2);
```

Comment:
Simple pattern: can we avoid the extra array: https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/24181/Garbage-Collection?anchor=%E2%9D%8C-**avoid**-using-%60string.split%60-on-simple-patterns?